File: apps/backend/src/EnterpriseAgentOs.Domain/Interfaces/Auth/ISessionRepository.cs
namespace EnterpriseAgentOs.Domain.Interfaces.Auth;

public interface ISessionRepository
{
    Task<SessionRecord> CreateAsync(Guid userId, string tokenHash, DateTime expiresAt, CancellationToken ct = default);
    Task<SessionRecord?> GetByTokenHashAsync(string tokenHash, CancellationToken ct = default);
    Task DeleteAsync(string tokenHash, CancellationToken ct = default);
    Task PurgeExpiredAsync(CancellationToken ct = default);
}

File: apps/backend-java/src/main/java/com/enterpriseagentos/backendjava/domain/features/management/ISessionRepository.java
package com.enterpriseagentos.backendjava.domain.features.management;

import java.math.BigDecimal;
import java.time.Instant;
import java.time.YearMonth;
import java.util.*;
import java.util.concurrent.CompletableFuture;
import com.enterpriseagentos.backendjava.domain.common.primitives.*;
import com.enterpriseagentos.backendjava.domain.common.services.*;
import com.enterpriseagentos.backendjava.domain.common.valueobjects.*;
import com.enterpriseagentos.backendjava.domain.events.*;
import com.enterpriseagentos.backendjava.domain.features.agents.*;
import com.enterpriseagentos.backendjava.domain.features.analytics.*;
import com.enterpriseagentos.backendjava.domain.features.management.*;
import com.enterpriseagentos.backendjava.domain.features.mcp.*;

public interface ISessionRepository {
    CompletableFuture<SessionRecord> createAsync(UUID userId, String tokenHash, Instant expiresAt);
    CompletableFuture<SessionRecord> getByTokenHashAsync(String tokenHash);
    CompletableFuture<Void> deleteAsync(String tokenHash);
    CompletableFuture<Void> purgeExpiredAsync();
    CompletableFuture<Void> deleteByUserIdAsync(UUID userId);
}
